src/Sulu/Bundle/AdminBundle/Resources/js/containers/Form/stores/ResourceFormStore.js

```diff
@@ -87,15 +88,163 @@ export default class ResourceFormStore extends AbstractFormStore implements Form
         });
     };
 
-    @action handleSchemaResponse = ([schema, jsonSchema]: [Schema, Object]) => {
+    handleSchemaResponse = ([schema, jsonSchema]: [Schema, Object]) => {
         this.validator = jsonSchema ? ajv.compile(jsonSchema) : undefined;
         this.pathsByTag = {};
 
-        this.schema = schema;
-        this.addMissingSchemaProperties();
-        this.setSchemaLoading(false);
+        this.removeObsoleteData(this.schema, schema, this.data);
```

Comment:
do we need the old schema in this method? 
maybe we could implement it similar to the `addMissingSchemaProperties` method and call it `removeUnknownSchemaProperties`?

Reply:
think i would also move the method to the `AbstractFormStore` if the comment above is possible